flytekit/types/directory/types.py

```diff
@@ -665,8 +667,7 @@ async def async_to_python_value(
 
         batch_size = get_batch_size(expected_python_type)
 
-        def _downloader():
-            return ctx.file_access.get_data(uri, local_folder, is_multipart=True, batch_size=batch_size)
+        _downloader = partial(_flyte_directory_downloader, ctx.file_access, uri, local_folder, batch_size)
```

Comment:
Looking at this again, I think we can partial `ctx.file_access` itself and not need to define `_flyte_directory_downloader`.

```python
_downloader = partial(ctx.file_access.get_data, url, local_folder, is_multipart=True, batch_size=batch_size)
```